### PR TITLE
Add KB30 keyboard

### PR DESCRIPTION
**Pull request by DOIO2022**
## Description

   

Add  kb30 keyboard

## QMK Pull Request 

https://github.com/qmk/qmk_firmware/pull/19395

## Checklist

- [x] The VIA support for this keyboard is **MERGED** in QMK master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have a V3 JSON version for this keyboard definition.**(MANDATORY)**
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

Thank you V3

### CODE REVIEW

**Comment by Xelus22** [comment]
Please fix your V2. it is not the same as V3.

**Comment by DOIO2022** [comment]
> 



> Please fix your V2. it is not the same as V3.

Hi, I have  fixed V2
Thank you!

**Comment by Xelus22** [comment]
Looks like something is wrong. :(

**Comment by DOIO2022** [comment]
@Xelus22  Can I only add V3?

**Comment by Xelus22** [comment]
yes you can

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by DOIO2022** [comment]
> https://aka.ms/AAb9ysg

@Xelus22   I've deleted it. Thank you!

File: v3/doio/kb30/kb30-01.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  "name": "DOIO",
+  "vendorId": "0xD010",
+  "productId": "0x3001",
+  "keycodes": ["qmk_lighting"],
+  "menus": ["qmk_rgb_matrix"],
```
Comment (Xelus22) [inline v3/doio/kb30/kb30-01.json:6]: https://github.com/qmk/qmk_firmware/pull/19395/files#diff-47d8ce7c695a8cadbba1922ca5ff8649ec5e62eec7b69c3e355635560df66f25R77-R120

Your default configuration does **not** enable all the RGB Matrix effects.

However the VIA default configuration does.

The user will experience the name of the effect does not matchup to the effect displayed on the board.
Please look at 
https://github.com/the-via/reader/blob/master/src/common-menus/qmk_rgb_matrix.ts

https://www.caniusevia.com/docs/built_in_menus

An example of another keyboard without all the animations defined is 
https://github.com/the-via/keyboards/blob/master/v3/horrortroll/handwired_k552/handwired_k552.json

**Review by Xelus22 — CHANGES_REQUESTED** [review]

File: v3/doio/kb30/kb30-01.json (hunk 1 above)
Comment (DOIO2022) [reply v3/doio/kb30/kb30-01.json]: @Xelus22  Hi,thank you for your advice,I've changed the file.

File: v3/doio/kb30/kb30-01.json [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                "Solid Reactive Wide",
+                "Solid Reactive Multi Wide",
+                "Solid Reactive Cross",
+                "Solid Reactive Multi Cross",
+                "Solid Reactive Nexus",
+                "Solid Reactive Multi Nexus",
+                "Spash",
+                "Multi Splash",
+                "Solid Splash",
+                "Solid Multi Splash"
```
Comment (Xelus22) [inline v3/doio/kb30/kb30-01.json:68]: image

You've enabled all of them, but your VIA does not match QMK firmware. You should delete the ones in the VIA json list that isnt enabled in QMK

**Review by Xelus22 — CHANGES_REQUESTED** [review]

File: v3/doio/kb30/kb30-01.json (hunk 2 above)
Comment (DOIO2022) [reply v3/doio/kb30/kb30-01.json]: @Xelus22  Hi, thank you for your help,  maybe this time is right.

**Pull request merged**